### ISSUE TITLE
torch.nn.BatchNorm expects positive eps in eval mode

### ISSUE THREAD

**Issue by bhushan23** [pytorch/pytorch#173337]
### 🐛 Describe the bug

torch 2.10.0 enforces `torch.nn.BatchNorm` to have positive eps https://github.com/pytorch/pytorch/blob/567eee90e8af4f51f9dba3034902ce4d0aa5ebd6/torch/nn/functional.py#L2843-L2844

In eval mode, `running_mean` and `running_var` are frozen and can be constant-folded and hence eps can be 0 

**Repro**
```python
import torch
import torch.nn as nn

batch_norm = nn.BatchNorm1d(
    num_features=512,
    eps=0,
    momentum=0.9
)

# eval mode
batch_norm.eval()

input_tensor = torch.randn(16, 512)  # (batch_size, num_features)
output = batch_norm(input_tensor)
```

There are couple of options here
1. Check `running_var` along with `eps` to avoid `divide-by-zero`
2. non-zero `eps` check only in case of training mode

option 1 is more general but comes with added complexity of checking `running_var`

### Versions

torch==2.10.0

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki
